### PR TITLE
JSON o/p format for the logs generated from entrypoint script

### PR DESCRIPTION
**Pull request by jayanthvn**
*Issue #907 , if available: JSON o/p format for the logs generated from entrypoint script

*Description of changes:*
dev-dsk-varavaj-2b-72f02457 % kubectl logs aws-node-2sq6w -n kube-system
{"level":"info","ts":"2020-07-01T02:21:56.104Z","caller":"entrypoint.sh","msg":"Copying CNI plugin binaries ... "}
{"level":"info","ts":"2020-07-01T02:21:56.120Z","caller":"entrypoint.sh","msg":"Starting IPAM daemon in the background ... "}
{"level":"info","ts":"2020-07-01T02:21:56.127Z","caller":"entrypoint.sh","msg":"Checking for IPAM connectivity ... "}
{"level":"info","ts":"2020-07-01T02:21:58.148Z","caller":"entrypoint.sh","msg":"Copying CNI plugin binary and config file ... "}
{"level":"info","ts":"2020-07-01T02:21:58.167Z","caller":"entrypoint.sh","msg":"Successfully copied CNI plugin binary and config file."}
{"level":"info","ts":"2020-07-01T02:21:58.168Z","caller":"entrypoint.sh","msg":"Foregrounding IPAM daemon ..."}

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by mogren** [comment]
I think I need to update #907 a bit. Having these log messages definitely helps us debug issues, so it would be better to have a function that logs messages in JSON format instead, so that it matches what zap outputs. Something like:
```
{"level":"info","ts":"2020-06-30T03:45:53.337Z","caller":"entrypoint.sh","msg":"Copying CNI plugin binary and config file ... "}
{"level":"info","ts":"2020-06-30T03:45:53.422Z","caller":"entrypoint.sh","msg":"Foregrounding IPAM daemon ... "}
```

**Comment by jayanthvn** [comment]
> I think I need to update #907 a bit. Having these log messages definitely helps us debug issues, so it would be better to have a function that logs messages in JSON format instead, so that it matches what zap outputs. Something like:
> 
> ```
> {"level":"info","ts":"2020-06-30T03:45:53.337Z","caller":"entrypoint.sh","msg":"Copying CNI plugin binary and config file ... "}
> {"level":"info","ts":"2020-06-30T03:45:53.422Z","caller":"entrypoint.sh","msg":"Foregrounding IPAM daemon ... "}
> ```

Thanks @mogren . It makes sense and I can look into it.

File: scripts/entrypoint.sh [hunk 1]
```diff
@@ -26,13 +26,22 @@ set -eu
...
 set -m
 
+function log_in_json() 
+{
+    FILENAME="${0##*/}"
+    LOGTYPE=$1
+    MSG=$2
+    TIMESTAMP=$(date +%FT%T.%3NZ)
+    echo '{"level":'"$LOGTYPE"',"ts":'"$TIMESTAMP"',"caller":'"$FILENAME"',"msg":'"$MSG"'}'
+}
```
[the remark this answers is not in the document]
Comment (jayanthvn) [reply scripts/entrypoint.sh:36]: Thanks  @nithu0115. Will address the comments.

[the remark this answers is not in the document]
Comment (jayanthvn) [reply scripts/entrypoint.sh:36]: Updated based on the CR comments -

dev-dsk-varavaj-2b-72f02457 % kubectl logs aws-node-cz56g -n kube-system
{"level":"info","ts":"2020-07-01T01:18:06.524Z","caller":"entrypoint.sh","msg":"Copying CNI plugin binaries ... "}
{"level":"info","ts":"2020-07-01T01:18:06.537Z","caller":"entrypoint.sh","msg":"Starting IPAM daemon in the background ... "}
{"level":"info","ts":"2020-07-01T01:18:06.543Z","caller":"entrypoint.sh","msg":"Successfully started IPAM daemon."}
{"level":"info","ts":"2020-07-01T01:18:06.544Z","caller":"entrypoint.sh","msg":"Checking for IPAM connectivity ... "}
{"level":"info","ts":"2020-07-01T01:18:08.561Z","caller":"entrypoint.sh","msg":"Successfully connected to IPAM daemon."}
{"level":"info","ts":"2020-07-01T01:18:08.562Z","caller":"entrypoint.sh","msg":"Copying CNI plugin binary and config file ..."}
{"level":"info","ts":"2020-07-01T01:18:08.580Z","caller":"entrypoint.sh","msg":"Successfully copied."}
{"level":"info","ts":"2020-07-01T01:18:08.581Z","caller":"entrypoint.sh","msg":"Foregrounding IPAM daemon ..."}

File: scripts/entrypoint.sh [hunk 2]
```diff
@@ -26,13 +26,22 @@ set -eu
 # turn on bash's job control
 set -m
 
+log_in_json() 
+{
+    FILENAME="${0##*/}"
```
Comment (mogren) [inline scripts/entrypoint.sh:31]: Hadn't seen `"${0##*/}"` before 😄

Comment (jayanthvn) [reply scripts/entrypoint.sh:31]: _From stack overflow_ - "for the variable $0, and the pattern '/', the two hashes mean from the beginning of the parameter, delete the longest (or greedy) match—up to and including the pattern. So, where $0 is the name of a file, eg., $HOME/documents/doc.txt, then the parameter would be expanded as: doc.txt" :D

**Pull request merged**